client/my-sites/checkout/composite-checkout/hooks/use-valid-checkout-back-url.ts

```diff
@@ -11,6 +12,7 @@ const getAllowedHosts = ( siteSlug?: string ) => {
 		'jetpack.com',
 		'jetpack.cloud.localhost',
 		'cloud.jetpack.com',
+		config( 'hostname' ),
```

Comment:
@Automattic/shilling hi 👋🏼  To allow people to go back to Stepper flows, we need to add WordPress.com (or Calypso.live) host names to allowed hosts. Are there any bad consequences to this?

Reply:
I don't think that would cause any problems. AFAIK this function is designed to prevent open redirects where a URL has a query string like `?redirectTo=malicious-site.org`. Since we trust domains that we own, I think that adding them is safe. They're not in here currently, I believe, because this function has only been run on the wordpress.com domain.

However, I am curious about the use-case. Is the Stepper not on wordpress.com? Note that `useValidCheckoutBackUrl()` is only used when a Jetpack user exits checkout without completing a purchase (eg: by removing the last item in their cart). How would this function prevent people from returning to the Stepper?

Reply:
Great question. Stepper is indeed under WordPress.com, but it's an independent app that doesn't share state with the rest of Calypso nor with checkout. This results in [this line](https://github.com/Automattic/wp-calypso/blob/4bd15986de292f2452a5ef844034865bc65e8450/client/my-sites/checkout/composite-checkout/components/wp-checkout.tsx#L291) not returning the correct path.

Reply:
Ah, interesting. Ok, feel free to add those domains. That ends up being similar to [this list](https://github.com/Automattic/wp-calypso/blob/0f7f7bdb6790b1db87ed2badbb062b67dd4b7bbb/client/my-sites/checkout/composite-checkout/lib/pending-page.ts#L212).